### PR TITLE
DBPW - Update sdk to get new package name

### PR DESCRIPTION
**Pull request by pcman312**
# Overview
Updates the sdk package to get the new `dbplugin` package name

# Design of Change
`go mod` and some import renames

# Related Issues/Pull Requests
https://github.com/hashicorp/vault/pull/10151

# Test Output

<details>

<summary>Test Output</summary>
START| _Acceptance
SKIP | _Acceptance (0.00s)
START| Client_CreateListGetDeleteRole
     | 2020/10/15 14:51:46 [DEBUG] POST http://127.0.0.1:61092/_xpack/security/role/role-name
     | 2020/10/15 14:51:46 [DEBUG] GET http://127.0.0.1:61092/_xpack/security/role/role-name
     | 2020/10/15 14:51:46 [DEBUG] DELETE http://127.0.0.1:61092/_xpack/security/role/role-name
PASS | Client_CreateListGetDeleteRole (0.00s)
START| Client_CreateGetDeleteUser
     | 2020/10/15 14:51:46 [DEBUG] POST http://127.0.0.1:61096/_xpack/security/user/user-name
     | 2020/10/15 14:51:46 [DEBUG] POST http://127.0.0.1:61096/_xpack/security/user/user-name/_password
     | 2020/10/15 14:51:46 [DEBUG] DELETE http://127.0.0.1:61096/_xpack/security/user/user-name
PASS | Client_CreateGetDeleteUser (0.00s)
START| Client_BadResponses
     | 2020/10/15 14:51:46 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/200-but-body-changed
     | 2020/10/15 14:51:46 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/404-not-found
     | 2020/10/15 14:51:46 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/500-mysterious-internal-server-error
     | 2020/10/15 14:51:46 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/500-mysterious-internal-server-error (status: 500): retrying in 1s (4 left)
     | 2020/10/15 14:51:47 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/500-mysterious-internal-server-error (status: 500): retrying in 2s (3 left)
     | 2020/10/15 14:51:49 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/500-mysterious-internal-server-error (status: 500): retrying in 4s (2 left)
     | 2020/10/15 14:51:53 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/500-mysterious-internal-server-error (status: 500): retrying in 8s (1 left)
     | 2020/10/15 14:52:01 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/503-unavailable
     | 2020/10/15 14:52:01 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/503-unavailable (status: 503): retrying in 1s (4 left)
     | 2020/10/15 14:52:02 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/503-unavailable (status: 503): retrying in 2s (3 left)
     | 2020/10/15 14:52:04 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/503-unavailable (status: 503): retrying in 4s (2 left)
     | 2020/10/15 14:52:08 [DEBUG] GET http://127.0.0.1:61100/_xpack/security/role/503-unavailable (status: 503): retrying in 8s (1 left)
PASS | Client_BadResponses (30.02s)
START| Elasticsearch
START|   Elasticsearch/test_type
START|   Elasticsearch/test_initialize
     | 2020/10/15 14:52:16 [DEBUG] GET http://127.0.0.1:61683/_xpack/security/role/vault-test
START|   Elasticsearch/test_new_user
     | 2020/10/15 14:52:16 [DEBUG] POST http://127.0.0.1:61683/_xpack/security/role/v-display-name-role-name-S22WeOrzH2lY9uggwzBZ-1602795136
     | 2020/10/15 14:52:16 [DEBUG] POST http://127.0.0.1:61683/_xpack/security/user/v-display-name-role-name-S22WeOrzH2lY9uggwzBZ-1602795136
     | 2020/10/15 14:52:16 [DEBUG] POST http://127.0.0.1:61683/_xpack/security/user/v-display-name-role-name-ErJDJxYiNFG52qSnGm3f-1602795136
START|   Elasticsearch/test_delete_user
     | 2020/10/15 14:52:16 [DEBUG] DELETE http://127.0.0.1:61683/_xpack/security/role/v-display-name-role-name-S22WeOrzH2lY9uggwzBZ-1602795136
     | 2020/10/15 14:52:16 [DEBUG] DELETE http://127.0.0.1:61683/_xpack/security/user/v-display-name-role-name-S22WeOrzH2lY9uggwzBZ-1602795136
     | 2020/10/15 14:52:16 [DEBUG] DELETE http://127.0.0.1:61683/_xpack/security/role/v-display-name-role-name-ErJDJxYiNFG52qSnGm3f-1602795136
     | 2020/10/15 14:52:16 [DEBUG] DELETE http://127.0.0.1:61683/_xpack/security/user/v-display-name-role-name-ErJDJxYiNFG52qSnGm3f-1602795136
START|   Elasticsearch/test_update_user
     | 2020/10/15 14:52:16 [DEBUG] POST http://127.0.0.1:61683/_xpack/security/user/fizz/_password
PASS | Elasticsearch (0.00s)
PASS |   Elasticsearch/test_type (0.00s)
PASS |   Elasticsearch/test_initialize (0.00s)
PASS |   Elasticsearch/test_new_user (0.00s)
PASS |   Elasticsearch/test_delete_user (0.00s)
PASS |   Elasticsearch/test_update_user (0.00s)
START| Elasticsearch_SecretValues
PASS | Elasticsearch_SecretValues (0.00s)
START| Integration_Container
START|   Integration_Container/test_init
     | 2020/10/15 14:52:49 [DEBUG] GET https://localhost:32784/_xpack/security/role/vault-test
START|   Integration_Container/test_create_user
     | 2020/10/15 14:52:49 [DEBUG] POST https://localhost:32784/_xpack/security/role/v-display-name-role-name-eh267TcPSI4BRDH7VuWx-1602795169
     | 2020/10/15 14:52:49 [DEBUG] POST https://localhost:32784/_xpack/security/user/v-display-name-role-name-eh267TcPSI4BRDH7VuWx-1602795169
     | 2020/10/15 14:52:49 [DEBUG] POST https://localhost:32784/_xpack/security/user/v-display-name-role-name-oEwpaX9YmpGZ4VpFDxV8-1602795169
START|   Integration_Container/test_delete_user
     | 2020/10/15 14:52:49 [DEBUG] DELETE https://localhost:32784/_xpack/security/role/v-display-name-role-name-eh267TcPSI4BRDH7VuWx-1602795169
     | 2020/10/15 14:52:49 [DEBUG] DELETE https://localhost:32784/_xpack/security/user/v-display-name-role-name-eh267TcPSI4BRDH7VuWx-1602795169
     | 2020/10/15 14:52:49 [DEBUG] DELETE https://localhost:32784/_xpack/security/role/v-display-name-role-name-oEwpaX9YmpGZ4VpFDxV8-1602795169
     | 2020/10/15 14:52:50 [DEBUG] DELETE https://localhost:32784/_xpack/security/user/v-display-name-role-name-oEwpaX9YmpGZ4VpFDxV8-1602795169
START|   Integration_Container/test_update_user
     | 2020/10/15 14:52:50 [DEBUG] POST https://localhost:32784/_xpack/security/user/vault/_password
PASS | Integration_Container (33.93s)
PASS |   Integration_Container/test_init (0.03s)
PASS |   Integration_Container/test_create_user (0.50s)
PASS |   Integration_Container/test_delete_user (0.24s)
PASS |   Integration_Container/test_update_user (0.33s)
PASS
PASS | github.com/hashicorp/vault-plugin-database-elasticsearch 65.362s
</details>

# Contributor Checklist
- [x] Docs not applicable here
- [x] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
- [x] Backwards compatible with dbplugin version 5

### CODE REVIEW

File: elasticsearch.go [hunk 1]
```diff
@@ -10,17 +10,17 @@ import (
...
 	multierror "github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/vault/api"
+	dbplugin "github.com/hashicorp/vault/sdk/database/dbplugin/v5"
 	"github.com/hashicorp/vault/sdk/database/helper/credsutil"
 	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
-	"github.com/hashicorp/vault/sdk/database/newdbplugin"
 )
 
-var _ newdbplugin.Database = &Elasticsearch{}
+var _ dbplugin.Database = &Elasticsearch{}
```
Comment (calvn) [inline elasticsearch.go:18]: Nit:
```suggestion
var _ dbplugin.Database = (*Elasticsearch)(nil)
```

**Review by calvn — APPROVED** [review]

File: vendor/modules.txt [hunk 2]
```diff
@@ -74,14 +74,15 @@ github.com/hashicorp/hcl/json/token
...
-# github.com/hashicorp/vault/sdk v0.1.14-0.20200930230837-9978ba802fcd
+# github.com/hashicorp/vault/sdk v0.1.14-0.20201015192012-a69ee0f65a28
 ## explicit
 github.com/hashicorp/vault/sdk/database/dbplugin
+github.com/hashicorp/vault/sdk/database/dbplugin/v5
+github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto
+github.com/hashicorp/vault/sdk/database/dbplugin/v5/testing
 github.com/hashicorp/vault/sdk/database/helper/credsutil
 github.com/hashicorp/vault/sdk/database/helper/dbutil
 github.com/hashicorp/vault/sdk/database/newdbplugin
```
Comment (tvoran) [inline vendor/modules.txt:85]: Is newdbplugin still being vendored? FWIW `go mod vendor` removes it for me on this branch.

**Review by tvoran — APPROVED** [review]

File: vendor/modules.txt (hunk 2 above)
Comment (pcman312) [reply vendor/modules.txt]: It is but shouldn't be. Good catch.

File: elasticsearch.go (hunk 1 above)
Comment (pcman312) [reply elasticsearch.go]: Done

**Review by tvoran — APPROVED** [review]

**Pull request merged**